File: src/main/java/lindar/acolyte/vo/AccessCredentials.java
package lindar.acolyte.vo;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class AccessCredentials {
    private String apiUrl;
    private String sessionCookie;

    private String username;
    private String password;
    
    public AccessCredentials(String apiUrl, String username, String password) {
        this.apiUrl = apiUrl;
        this.username = username;
        this.password = password;
    }
    
    public AccessCredentials(String sessionCookie) {
        this.sessionCookie = sessionCookie;
    }
}

File: src/main/kotlin/lindar/acolyte/vo/AccessCredentials.kt
package lindar.acolyte.vo


data class AccessCredentials (
        val apiUrl: String,
        val username: String? = null,
        val password: String? = null,
        val sessionCookie: String? = null,
        val authHeader: String? = null
) {
    companion object {
        @JvmStatic fun withUsernameAndPassword(apiUrl: String, username: String, password: String): AccessCredentials {
            return AccessCredentials(apiUrl = apiUrl, username = username, password = password)
        }
        @JvmStatic fun withAuthHeader(apiUrl: String, authHeader: String): AccessCredentials {
            return AccessCredentials(apiUrl = apiUrl, authHeader = authHeader)
        }
        @JvmStatic fun withSessionCookie(apiUrl: String, sessionCookie: String): AccessCredentials {
            return AccessCredentials(apiUrl = apiUrl, sessionCookie = sessionCookie)
        }
    }
}